templates/etc/check_mk/multisite.d/wato/users.mk.j2

```diff
@@ -0,0 +1,30 @@
+{% for user in checkmk_server__multisite_users|list|d([]) %}
+{#
+ # Strip Ansible-only properties from user definitions
+ #}
+{%   for property in checkmk_server__ansible_user_properties|d([]) %}
+{%     if property in checkmk_server__multisite_users[user]|list %}
+{%       set _ = checkmk_server__multisite_users[user].pop(property) %}
+{%     endif %}
+{%   endfor %}
+{#
+ # Make sure default arguments are defined
+ #}
+{%   for key, value in checkmk_server__multisite_user_defaults.iteritems() %}
+{%     if user not in checkmk_server_fact_local_users|d({}) and key not in checkmk_server__multisite_users[user]|d({}) %}
+{%       set _ = checkmk_server__multisite_users[user].update({key: value}) %}
+{%     endif %}
+{%   endfor %}
+{% endfor %}
+{#
+ # Combine fact users with multisite users
+ #}
+{% set users = checkmk_server_fact_local_users|d({}) | combine(checkmk_server__multisite_users, recursive=True) %}
+# Written by Multisite UserDB
```

Comment:
As Check_MK will re-generate the file (e.g. after syncing LDAP users; see [Automatic User Synchronization](https://mathias-kettner.com/checkmk_multisite_ldap_integration.html#Automatic%20User%20Synchronization)), I try to avoid any non-functional changes to this file as they won't be picked up by Check_MK and therefore overwritten again anyway. I try to write out the file identical as Check_MK would to avoid any unnecessary Ansible task executions. This should hopefully work, even when someone manually configures a user via WATO or a LDAP sync happened. I know that's a risky plan because any upstream change in this area could break the template, but let's see how it goes for now. I'll still add a comment to the template to explain this.